src/middlewared/middlewared/plugins/network.py

```diff
@@ -584,23 +608,45 @@ def iface_extend(self, iface_state, configs, fake=False):
 
         await self.__save_datastores()
 
-        if data['type'] == 'LINK_AGGREGATION':
-            name = data.get('name')
-            if not name:
-                next_lagg = 0
-                laggs = [
-                    i['int_interface']
-                    for i in await self.middleware.call(
-                        'datastore.query',
-                        'network.interfaces',
-                        [('int_interface', '^', 'lagg')],
-                    )
-                ]
-                while f'lagg{next_lagg}' in laggs:
-                    next_lagg += 1
-                name = f'lagg{next_lagg}'
+        async def get_next(prefix, start=0):
```

Comment:
Can we define this in the outer scope?

Reply:
I don't like when functions are defined inside functions because:
* It's unclear what inner function has captured to it's scope. The longer it's body the more unclear it is
* You can't unit-test inner function or reuse it somewhere else
* It just looks like a mess

And in this specific situation we benefit nothing from it.